src/runtime/runtime.cs

```diff
@@ -328,6 +320,29 @@ internal static void Initialize(bool initSigs = false)
             AssemblyManager.UpdatePath();
         }
 
+        private static IntPtr Get_PyObject_NextNotImplemented()
+        {
+            IntPtr globals = PyDict_New();
+            if (PyDict_SetItemString(globals, "__builtins__", PyEval_GetBuiltins()) != 0)
+            {
+                XDecref(globals);
+                throw new PythonException();
+            }
+            const string code = "class A(object): pass";
+            IntPtr res = PyRun_String(code, (IntPtr)RunFlagType.File, globals, globals);
+            if (res == IntPtr.Zero)
+            {
+                XDecref(globals);
+                throw new PythonException();
```

Comment:
You might want to construct an instance of `PythonException` before doing `XDecref`, otherwise last error could potentially be overwritten by `XDecref` with something new.